sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVectorCore.java

```diff
@@ -0,0 +1,169 @@
+
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids;
+
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+
+/**
+ * A GPU accelerated version of the Spark ColumnVector.
+ * Most of the standard Spark APIs should never be called, as they assume that the data
+ * is on the host, and we want to keep as much of the data on the device as possible.
+ * We also provide GPU accelerated versions of the transitions to and from rows.
+ */
+public final class RapidsHostColumnVectorCore extends ColumnVector {
+
+  /**
+   * Get the underlying host cudf columns from the batch.  This does not increment any
+   * reference counts so if you want to use these columns after the batch is closed
+   * you will need to do that on your own.
+   */
+  public static ai.rapids.cudf.HostColumnVectorCore[] extractBases(ColumnarBatch batch) {
+    int numColumns = batch.numCols();
+    ai.rapids.cudf.HostColumnVectorCore[] vectors = new ai.rapids.cudf.HostColumnVectorCore[numColumns];
+    for (int i = 0; i < vectors.length; i++) {
+      vectors[i] = ((RapidsHostColumnVector)batch.column(i)).getBase();
```

Comment:
Is this cast still correct? Is this API even used?

Reply:
I will remove these apis just as I did with copytodevice. We dont really need these for now and we can always go back and add them

Reply:
done.